agent.js

```diff
@@ -26,64 +29,70 @@ export class Agent {
                 if (username === this.name) return;
                 console.log('received message from', username, ':', message);
     
-                this.respond(username, message);
-                this.history.save();
-            });
-    
-            this.bot.on('finished_executing', () => {
-                setTimeout(() => {
-                    if (!this.coder.executing) {
-                        // return to default behavior
-                    }
-                }, 10000);
+                this.history.add(username, message);
+                this.handleMessage();
             });
 
-            if (autostart)
-                this.respond('system', 'Agent process restarted. Notify the user and decide what to do.');
-            
+            if (init_message) {
+                this.history.add('system', init_message);
+                this.handleMessage();
```

Comment:
seems confusing/dangerous to add to history then call handleMessage. Why can't we do that inside the function?

Reply:
We would need to only add it the first time the function is called and not during any of the recursions. I could create a wrapper function that adds to history and then calls an internalHandleMessage function.